table/table.go

```diff
@@ -266,7 +266,11 @@ func (m *Model) UpdateViewport() {
 // SelectedRow returns the selected row.
 // You can cast it to your own implementation.
 func (m Model) SelectedRow() Row {
-	return m.rows[m.cursor]
+	// Check if the cursor is within bounds else return  nil
+	if m.cursor < len(m.rows) && m.cursor >= 0 {
+		return m.rows[m.cursor]
+	}
+	return nil
```

Comment:
Like this?
```suggestion
	if !(m.cursor >= 0 && m.cursor < len(m.rows)) {
		return nil
	}
	return m.rows[m.cursor]
```

Reply:
I'd probably go with:

```go
if m.cursor < 0 || m.cursor >= len(m.rows) {
    return nil
}

return m.rows[m.cursor]
```

Should be a little more digestible for future readers.

Reply:
I see I've added a commit with the suggested change. Thank you for the comments. (:

Reply:
And sorry for the mess with all the commits 😅